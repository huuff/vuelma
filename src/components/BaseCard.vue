<template>
  <div class="card">
    <header class="card-header">
      <slot name="header">
        <p v-if="headerTitle" class="card-header-title">
          {{ headerTitle }}
        </p>
        <div v-if="icon" class="card-header-icon">
          <span class="icon">
            <font-awesome-icon :icon="icon" aria-hidden></font-awesome-icon>
          </span>
        </div>
      </slot>
    </header>
    <div v-if="image" class="card-image">
      <figure class="image">
        <img :src="image">
      </figure>
    </div>
    <div v-if="$slots.default" class="card-content">
      <slot></slot>
    </div>
    <footer v-if="$slots.footer || $slots.footerItems" class="card-footer"> 
      <slot name="footer"></slot>
      <classed-slot :children-class="'card-footer-item'">
        <slot name="footerItems"></slot>
      </classed-slot>
    </footer>
  </div>
</template>

<script setup lang="ts">
import { useSlots } from 'vue';
import { FontAwesomeIconName } from '@/types/fontawesome-icon-name';
import ClassedSlot from '@/components/slots/ClassedSlot';

const props = defineProps<{
  headerTitle?: string;
  image?: string;
  icon?: FontAwesomeIconName;
}>();

const slots = useSlots();

if (slots.footer && slots.footerItems) {
  throw new Error('The card component cannot be used with both a footer and footerItems!');
}

if (slots.header && (slots.headerTitle || slots.icon)) {
  throw new Error('The card component cannot be used with both a header and a headerTitle or an icon!');
}
</script>
