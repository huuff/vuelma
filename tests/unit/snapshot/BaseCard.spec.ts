import { mount } from "@vue/test-utils";
import BaseCard from "@/components/BaseCard.vue";

test("BaseCard.vue snapshot", () => {
  const wrapper = mount(BaseCard, {
    props: {
      titleText: "Card Title",
      image: "testimg",
      icon: "heart"
    },
    global: {
      stubs: [ "font-awesome-icon" ],
    },
    slots: {
      default: `
        <p>Content of the card body</p>
      `,
      footerItems: `
        <a>Footer item 1</a>
        <a>Footer item 2</a>
        <a>Footer item 3</a>
      `
    }
  });

  expect(wrapper.element).toMatchSnapshot();
});
