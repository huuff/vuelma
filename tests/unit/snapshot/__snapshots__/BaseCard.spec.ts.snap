// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseCard.vue snapshot 1`] = `
<div
  class="card"
>
  <header
    class="card-header"
  >
    
    <p
      class="card-header-title"
    >
      Card Title
    </p>
    <div
      class="card-header-icon"
    >
      <span
        class="icon"
      >
        <font-awesome-icon-stub
          aria-hidden=""
          icon="heart"
        />
      </span>
    </div>
    
  </header>
  <div
    class="card-image"
  >
    <figure
      class="image"
    >
      <img
        src="testimg"
      />
    </figure>
  </div>
  <div
    class="card-content"
  >
    
    <p>
      Content of the card body
    </p>
    
  </div>
  <footer
    class="card-footer"
  >
    
    
    
    
    <a
      class="card-footer-item"
    >
      Footer item 1
    </a>
    <a
      class="card-footer-item"
    >
      Footer item 2
    </a>
    <a
      class="card-footer-item"
    >
      Footer item 3
    </a>
    
    
  </footer>
</div>
`;
